src/request_tracing.cpp

```diff
@@ -196,34 +271,40 @@ void RequestTracing::on_log_request() {
   add_script_tags(main_conf_->tags, request_, *request_span_);
   add_upstream_name(request_, *request_span_);
 
-  // When datadog_operation_name points to a variable and it can be initialized
-  // or modified at any phase of the request, so set the span operation name at
-  // request exit phase, which will take the latest value of the variable
-  // pointed to by the datadog_operation_name directive
+  // When datadog_operation_name points to a variable, then it can be
+  // initialized or modified at any phase of the request, so set the span
+  // operation name at request exit phase, which will take the latest value of
+  // the variable pointed to by the datadog_operation_name directive. Similarly
+  // with resource name.
   auto core_loc_conf = static_cast<ngx_http_core_loc_conf_t *>(
       ngx_http_get_module_loc_conf(request_, ngx_http_core_module));
-  request_span_->SetOperationName(get_request_operation_name(request_, core_loc_conf, loc_conf_));
-  request_span_->SetTag("resource.name", get_request_resource_name(request_, loc_conf_));
+  request_span_->set_name(get_request_operation_name(request_, core_loc_conf, loc_conf_));
+  request_span_->set_resource_name(get_request_resource_name(request_, loc_conf_));
```

Comment:
Earlier I introduced `$datadog_location` and set it as the default resource name. That was problematic because customers wanted the entire URL instead. That will be problematic because the set of URLs might be large.

Let's kick it down the road. When we revisit it, I think it will be best to put the login into the nginx module, since it knows more about the meaning of the resource name than does the core library.